### ISSUE TITLE
[MapCard] Error: Can't resolve 'react/jsx-runtime'

### ISSUE THREAD

**Issue by kevinsperrine** [carbon-design-system/carbon-addons-iot-react#2421]
### What package is this for?

- [X] React
- [ ] Angular

### Describe the bug
The error below is the result of the react-dnd upgrade to 14.0.2 for the new drag-and-drop features of the MapCard. 

```
ERROR in ./node_modules/react-dnd/dist/esm/core/DndProvider.js
Module not found: Error: Can't resolve 'react/jsx-runtime' in '/home/sls/git/maximo-app-framework/graphite-alt-repos/graphite-mas/packages/react-components/node_modules/react-dnd/dist/esm/core'
 @ ./node_modules/react-dnd/dist/esm/core/DndProvider.js 17:0-48 54:9-13
 @ ./node_modules/react-dnd/dist/esm/core/index.js
 @ ./node_modules/react-dnd/dist/esm/index.js
 @ ./node_modules/carbon-addons-iot-react/lib/hooks/useDNDProviderElement.js
 @ ./node_modules/carbon-addons-iot-react/lib/index.js
 @ ./src/ui/common/Loading.js
 @ ./src/ui/TextInput/TextInput.js
 @ ./src/ui/Dialog/DialogController.story.js
 @ ./src/ui sync \.story\.js$
 @ ./config/storybook/config.js
 @ multi ./node_modules/@storybook/core/dist/server/common/polyfills.js ./node_modules/@storybook/core/dist/server/preview/globals.js ./config/storybook/config.js ./node_modules/webpack-hot-middleware/client.js?reload=true
```

Currently, carbon-addons-iot-react has a peer-dependency of `react@^16.8.6`, but react-dnd (since version 13) requires `react@>=16.14`. Technically, `react@^16.8.6` as a peer-dependency is loose enough to also match `react@16.14.0`, and b/c the `@storybook/angular` package has `react@16.14.0` as a dependency, our dev builds have been using `react@16.14.0` for a couple of months. See PR #2215.

![Screen Shot 2021-06-18 at 11 55 41 AM](https://user-images.githubusercontent.com/609466/122587702-26060800-d02c-11eb-8c98-a99dfcb0987e.png)

```
"@storybook/angular@^6.2.3":
  version "6.2.9"
  resolved "https://registry.yarnpkg.com/@storybook/angular/-/angular-6.2.9.tgz#531280cc87600bebb5833df611948b7e0cb27cc6"
  integrity sha512-/MFYooBL1LfGQZ+yeCDQi8n7oYS0JqXeEHO3pmtgCvWNC76pcHmFMmYgjrYlAYFHiTcv59PLWsLX9Q3VdUrhSg==
  dependencies:
    "@storybook/addons" "6.2.9"
    "@storybook/api" "6.2.9"
    "@storybook/core" "6.2.9"
    "@storybook/core-common" "6.2.9"
    "@storybook/node-logger" "6.2.9"
    "@types/webpack-env" "^1.16.0"
    autoprefixer "^9.8.6"
    core-js "^3.8.2"
    fork-ts-checker-webpack-plugin "^4.1.6"
    global "^4.4.0"
    postcss "^7.0.35"
    postcss-loader "^4.2.0"
    raw-loader "^4.0.2"
    react "16.14.0"
    react-dom "16.14.0"
    read-pkg-up "^7.0.1"
    regenerator-runtime "^0.13.7"
    sass-loader "^10.1.0"
    strip-json-comments "3.1.1"
    ts-dedent "^2.0.0"
    ts-loader "^8.0.14"
    tsconfig-paths-webpack-plugin "^3.3.0"
    util-deprecate "^1.0.2"
    webpack "4"
```

We have a few options on how to tackle this:

- Remove the drag-and-drop features from the MapCard
- Rollback react-dnd to version 11.x and implement the drag-and-drop features of the MapCard differently
- Immediately Bump the `carbon-addons-iot-react` react peer-dependency to `react@16.14.0`
- Leave this bump for a v3 release and move the drag-and-drop MapCard features into that branch.

### To Reproduce

- Create a react app using react version 16.8.6
- add carbon-addon-iot-react next.28 as a dependency
- build the project.

### Expected behavior

Storybook to build without any issues.

### Environment/versions:

- OS: [e.g. MacOS, Windows]
- Browser: [e.g. chrome, safari]
- `carbon-addons-iot-react` version: [e.g. `v2.60.0`]

### Additional context

react-dnd was upgraded here as part of this PR https://github.com/carbon-design-system/carbon-addons-iot-react/pull/2380

**Comment by bjornalm** [comment]
Ouch.

I would assume upgrading would be easiest, since we apparently already have been running 16.14.0 while testing and developing for a while. 

We could also downgrade and move the dnd-examples to the v3 branch, it is pretty well isolated from a code perspective since most of it was just new stories. I guess that if someone desperately needs it before v3 we can leave the new drop-ref in the MapCard and make the changes to the `useDNDProviderElement.jsx` conditional to the react-version and they can build the library with their own dependencies (i.e. dnd 14.0.2 and react 16.14.0)

**Comment by kevinsperrine** [comment]
@davidicus We didn't really define a plan of action on this one. It seems moot now that consumers will be upgrading to React 17 in their next update, right?

**Comment by davidicus** [comment]
Let's upgrade to the same version that is in the main package.json resolution field (`16.4.x`). Since this has been published for a while and most teams have already upgraded their React version I think we can leave the offending component as is. We should pin our react versions so we do not have these issues in the future.

**Comment by kevinsperrine** [comment]
closing as this was fixed by #2651.
